src/components/Toast/index.tsx

```diff
@@ -49,49 +49,64 @@ const ToastItemDiv = styled.div`
     }
   }
 `
-const ToastItem = ({
-  data,
-  disappearDuration,
-  willLeave,
-}: {
-  data: any
-  disappearDuration?: number
-  willLeave: Function
-}) => {
-  const [opacity, setOpacity] = useState(1)
+
+const ANIMATION_DISAPPEAR_TIME = 2000
+const MAX_FRAME: number = (ANIMATION_DISAPPEAR_TIME / 1000) * 40 // suppose fps = 40
+const DEFAULT_TOAST_DURATION = 3000
+
+export const useTimeout = (callback: () => void, clearFunction: () => void, delay: number) => {
+  const savedCallback = useRef(() => {})
+  const clearCallback = useRef(() => {})
+  useEffect(() => {
+    savedCallback.current = callback
+    clearCallback.current = clearFunction
+  })
   useEffect(() => {
-    let animationFun: any = null
-    const requestAnimationFrame = window.requestAnimationFrame || window.webkitRequestAnimationFrame
-    animationFun = setTimeout(() => {
-      const maxFrame: number = ((disappearDuration || 2000) / 1000) * 40 // suppose fps = 40
+    const tick = () => {
+      savedCallback.current()
+    }
+    const listener = setTimeout(tick, delay)
```

Comment:
It seems that the `tick` is redundant, `const listener = setTimeout(savedCallback.current, delay)` is enough.

Reply:
OK. I see.